woocommerce/woocommerce
# Update wc admin to React 18 and get Storybook working

### Changes proposed in this Pull Request:

This PR has the goal of getting the `@woocommerce/components` Storybook working properly again. To do that we update React in `@woocommerce/admin-library` to ensure that it uses React 18. This is not actually a runtime change because in the admin we have to use the bundled React version anyway.

I have opted to ts-ignore the ~89 type errors and leave to teams working on that code to fix. One that comes up a lot is that our buttons pass deprecated options that for some reason in the Gutenberg types are not added to the prop types even though they're still valid (deprecated not removed).

I've used `syncpack` to target the minimal set of packages to update to React 18, to avoid this PR causing widespread further type issues, but in future we can remove this package targeting as we upgrade the other remaining packages to React 18. 


### How to test the changes in this Pull Request:

1. CI should pass
2. A smoke test of wc admin functionality would be great. I went hunting for whitescreen issues but didn't see any. In theory as I said no runtime changes should happen but best to check
3. Try out the storybook. After `pnpm i`, `cd` to `tools/storybook` and run `pnpm storybook`.
4. Open the storybook at `localhost:6007` and click on components in the left sidebar and see them render without errors.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message
Update `@woocommerce/admin-library` to React 18 and fix `@woocommerce/components` Storybook.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
